vendor/processors/stripe_processor.py

```diff
@@ -198,6 +268,197 @@ def build_subscription(self, subscription, payment_method_id):
             'default_payment_method': payment_method_id,
             'metadata': {'site': self.invoice.site}
         }
+    
+    def build_subscription(self, customer_id, items, payment_id, offer):
+        subscription_data = {
+            'customer': customer_id,
+            'items': [{'price': item.id} for item in items],
+            'default_payment_method': payment_id,
+            'trial_period_days': None if offer.term_details['trial_days'] < 1 else offer.term_details['trial_days'],
+            'metadata': {'site': offer.site.pk}
+
+        }
+
+        subscription = self.stripe_create_object(self.stripe.Subscription, subscription_data)
+        
+        return subscription
+
+    ##########
+    # Stripe & Vendor Objects Synchronization
+    ##########
+
+    def get_stripe_customers(self):
+        """
+        Returns all Stripe created customers
+        """
+        query = self.build_search_query([{
+            'key_name': 'site',
+            'key_value': self.site.pk,
+            'field_type': 'metadata'
+        }])
+        customer_search = self.stripe_query_object(self.stripe.Customer, {'query': query})
+        if customer_search:
+            return customer_search['data']
+        return None
+
+    def get_stripe_customers_in_vendor(self, customers):
+        """
+        Returns all vendor customers who have been created as Stripe customers
+        """
+        email_list = [customer_obj['email'] for customer_obj in customers]
+        stripe_users = CustomerProfile.objects.filter(user__email__in=email_list)
+        return stripe_users
+
+    def get_stripe_customers_not_in_vendor(self, customers):
+        """
+        Returns all vendor customers who have not been created as Stripe customers
+        """
+
+        email_list = [customer_obj['email'] for customer_obj in customers]
+        stripe_users = CustomerProfile.objects.exclude(user__email__in=email_list)
+        return stripe_users
+
+    def get_customers_to_check(self):
+        stripe_customers = self.get_stripe_customers()
+        stripe_customers_in_vendor = self.get_stripe_customers_in_vendor(stripe_customers)
+        stripe_customers_not_in_vendor = self.get_stripe_customers_not_in_vendor(stripe_customers)
+        return stripe_customers_in_vendor, stripe_customers_not_in_vendor
+
+    def sync_customers(self):
+        stripe_customers_in_vendor, stripe_customers_not_in_vendor = self.get_customers_to_check()
```

Comment:
Eg here it is a loot easier to read if you seperate them into the three lines. the `get_customers_to_check` function is not really saving you any lines.


Also remeber that a function should only do one thing.